src/app/helpers/filters/filters.ts

```diff
@@ -0,0 +1,64 @@
+import { InnerFilter } from "@baw-api/baw-api.service";
+import { Writeable } from "@helpers/advancedTypes";
+import { isInstantiated } from "@helpers/isInstantiated/isInstantiated";
+import { AbstractModel } from "@models/AbstractModel";
+
+/**
+ * Add a filter to the root of a filter in an "and" expression
+ *
+ * @param filter The current existing filter conditions
+ * @param additionalFilter A new condition to add to the existing filter
+ * @returns A new filter which satisfies the intersection between the two filters
+ */
+export function filterAnd<T extends AbstractModel>(
+  filter: InnerFilter,
+  additionalFilter: InnerFilter
+): InnerFilter<Writeable<T>> {
+  if (!additionalFilter) {
+    return filter;
+  }
+
+  // because the api is expecting multiple conditions with the and combinator condition
+  // if there are multiple conditions, we need to wrap the filter in an and: [] combinator
+  // if there is only a single condition, the filter should be returned without the and block
+  if (!filter) {
+    return additionalFilter;
+  } else {
+    // if the current filter already contains an "and" conditional block, the additional filter should be added to the existing "and" block
+    // otherwise, the current and additional filters should be wrapped in an "and" conditional block together
+    return {
+      and: filter.and
+        ? [...filter.and as InnerFilter[], additionalFilter]
+        : [filter, additionalFilter],
+    };
+  }
+}
+
+/**
+ * Adds a new filter to an existing filter that filters by a model's id
+ *
+ * @param key The key identifier of the model, most likely the type
+ * @example key = "projects"
+ * @param model The model to filter by
+ * @param currentFilter (optional) A filter that model condition will be added to. If no filter is provided, a new filter will be created
+ * @returns A new filter with the model condition added
+ */
+export function filterModel<T extends AbstractModel, U extends AbstractModel>(
+  key: string,
+  model: T,
+  currentFilter?: InnerFilter
+): InnerFilter<Writeable<U>> {
+  // all model filters condition on the id attribute. While it is very rare for a model to not have an id, it is possible
+  // this bailout is typically evoked if the model is undefined
+  if (!isInstantiated(model?.id)) {
+    return currentFilter;
```

Comment:
i would suggest returning an empty inner filter e.g. `{ }` denotes that no filters are applied

Reply:
based on what you said this doesn't need to change because tests are artificially creating undefined filters